### PR TITLE
docs(Fx132): HTTP/2 Server Push is now deactivated by default in most major browsers

### PR DESCRIPTION
**Pull request by bsmth**
HTTP/2 Server Push is being unshipped everywhere. I'm checking everywhere we mention "Server Push" to make sure we're not recommending it anymore.

https://groups.google.com/a/mozilla.org/g/dev-platform/c/vU9hJg343U8/m/4cZsHz7TAQAJ

__Related issues and pull requests:__

- [x] Parent issue: https://github.com/mdn/content/issues/36116
- [x] Relnote: https://github.com/mdn/content/pull/36148

### CODE REVIEW

File: files/en-us/glossary/http_2/index.md [hunk 1]
```diff
@@ -8,7 +8,10 @@ page-type: glossary-definition
 
 **HTTP/2** is a major revision of the [HTTP network protocol](/en-US/docs/Web/HTTP).
 
-The primary goals for HTTP/2 are to reduce {{glossary("latency")}} and head-of-line blocking by enabling full request and response multiplexing, minimize protocol overhead via efficient compression of HTTP header fields, and add support for request prioritization and server push.
+The primary goals for HTTP/2 are to reduce {{glossary("latency")}} and head-of-line blocking by enabling full request and response multiplexing, minimize protocol overhead via efficient compression of HTTP header fields (HPACK), and support for request prioritization.
+
+HTTP/2 also introduced a mechanism called Server Push, which allowed a server to send resources to a client in anticipation that the client would need them very soon.
+Server Push proved tricky to implement in practice, and has been removed from most major browser engines, with methods such as [`rel="preload"`](/en-US/docs/Web/HTML/Attributes/rel/preload) and {{HTTPStatus("103", "103 Early hints")}} as alternatives.
```
Comment (hamishwillee) [inline files/en-us/glossary/http_2/index.md:14]:
```suggestion
Server Push proved tricky to implement in practice, and has been removed from most major browser engines.
It has been replaced by alternative methods such as [`rel="preload"`](/en-US/docs/Web/HTML/Attributes/rel/preload) and {{HTTPStatus("103", "103 Early hints")}}
```

File: files/en-us/learn/common_questions/web_mechanics/what_is_a_web_server/index.md [hunk 2]
```diff
@@ -83,9 +83,11 @@ A {{Glossary("Protocol")}} is a set of rules for communication between two compu
 - Stateless
   - : Neither the server nor the client remember previous communications. For example, relying on HTTP alone, a server can't remember a password you typed or remember your progress on an incomplete transaction. You need an application server for tasks like that. (We'll cover that sort of technology in other articles.)
 
-HTTP provides clear rules for how a client and server communicate. We'll cover HTTP itself in a [technical article](/en-US/docs/Web/HTTP) later. For now, just be aware of these things:
+HTTP provides clear rules for how a client and server communicate.
+If you want to learn more, you can read the [HTTP documentation](/en-US/docs/Web/HTTP).
+For now, there are a few things to keep in mind:
 
-- Usually only _clients_ make HTTP requests, and only to _servers_. Servers _respond_ to a _client_'s HTTP request. A server can also populate data into a client cache, in advance of it being requested, through a mechanism called [server push](https://en.wikipedia.org/wiki/HTTP/2_Server_Push).
+- _Clients_ make HTTP requests to _servers_. Servers _respond_ to a _client_'s HTTP request.
```
Comment (hamishwillee) [inline files/en-us/learn/common_questions/web_mechanics/what_is_a_web_server/index.md:90]: Losing the information about server push good. I don't think there are other _HTTP_ mechanisms for a server to push information to a client without their first being a request.

Though FYI it is possible for a server to include information in a response that was not requested which "somewhat" does the same thing (i.e. server provides a hint to client in any response that is unrelated to the request, triggering behaviour that looks a bit like a push). Upshot not worth mentioning here.

File: files/en-us/web/http/headers/save-data/index.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ content to reduce the data downloaded such as smaller image and video resources,
 different markup and styling, disabled polling and automatic updates, and so on.
 
 > [!NOTE]
-> Disabling HTTP/2 Server Push ({{RFC("7540", "Server Push", "8.2")}}) might be desirable too for reducing data downloads.
+> Disabling HTTP/2 Server Push ({{RFC("7540", "Server Push", "8.2")}}) may reduce data downloads, although this feature is no longer supported by default in most major browser engines.
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/save-data/index.md:46]:
```suggestion
> Disabling HTTP/2 Server Push ({{RFC("7540", "Server Push", "8.2")}}) may reduce data downloads.
> Note that this feature is no longer supported by default in most major browser engines.
```

**Review by hamishwillee — APPROVED** [review]
A couple of "to my taste" changes. Approving so you can merge or accept if you like.

Note that I only reviewed the changes that were made - I haven't checked if there are other mentions of server push that should be updated.

File: files/en-us/learn/common_questions/web_mechanics/what_is_a_web_server/index.md (hunk 2 above)
Comment (bsmth) [reply files/en-us/learn/common_questions/web_mechanics/what_is_a_web_server/index.md:90]: Right, I think for the target audience of this doc, we can keep it straightforward.

File: files/en-us/glossary/http_2/index.md [hunk 4]
```diff
@@ -8,7 +8,11 @@ page-type: glossary-definition
 
 **HTTP/2** is a major revision of the [HTTP network protocol](/en-US/docs/Web/HTTP).
 
-The primary goals for HTTP/2 are to reduce {{glossary("latency")}} and head-of-line blocking by enabling full request and response multiplexing, minimize protocol overhead via efficient compression of HTTP header fields, and add support for request prioritization and server push.
+The primary goals for HTTP/2 are to reduce {{glossary("latency")}} and head-of-line blocking by enabling full request and response multiplexing, minimize protocol overhead via efficient compression of HTTP header fields (HPACK), and support for request prioritization.
+
+HTTP/2 also introduced a mechanism called Server Push, which allowed a server to send resources to a client in anticipation that the client would need them very soon.
+Server Push proved tricky to implement in practice, and has been removed from most major browser engines.
+It has been replaced by alternative methods such as [`rel="preload"`](/en-US/docs/Web/HTML/Attributes/rel/preload) and {{HTTPStatus("103", "103 Early hints")}}
```
Comment (bsmth) [inline files/en-us/glossary/http_2/index.md:15]:
```suggestion
It has been replaced by alternative methods such as [`rel="preload"`](/en-US/docs/Web/HTML/Attributes/rel/preload) and {{HTTPStatus("103", "103 Early hints")}}.
```

**Comment by bsmth** [comment]
Thanks, Hamish!

> I haven't checked if there are other mentions of server push that should be updated.

I can't find any through code search, so I believe this is everything. I'll merge shortly 👍🏻

**Pull request merged**